tests/unit/test_plugins_loader.py

```diff
@@ -230,5 +230,23 @@ def test_load_plugins_none_loaded(mocker, plugins, exp_inited_plugins):
     mock_iter_entry_points.return_value = []
 
     loaded_config = {'core': {}}
-    inited_names, loaded_plugins = plugins_loader.load_plugins(loaded_config)
-    assert [] == loaded_plugins == inited_names
+    inited_names, loaded_plugins, errors = plugins_loader.load_plugins(
+        loaded_config)
+    assert [] == loaded_plugins == inited_names == errors
+
+
+def test_load_plugins_exceptions(plugins, exp_inited_plugins, loaded_config,
+                                 mock_iter_entry_points, plugin_exc_mock,
+                                 mocker, monkeypatch):
+    """Loading plugin exceptions are returned."""
+    names = ['one.plugin', 'two.plugin']
+    inited_plugins_mock = mocker.MagicMock(
+        plugins_loader._init_plugins, autospec=True)
+
+    exc = [('bad.plugin', plugin_exc_mock)]
+    inited_plugins_mock.return_value = names, inited_plugins_mock, exc
+    monkeypatch.setattr(plugins_loader, '_init_plugins', inited_plugins_mock)
+
+    inited_names, loaded_plugins, errors = plugins_loader.load_plugins(
+        loaded_config)
+    assert 1 == len(errors)
```

Comment:
Should `errors[0] == plugin_exc_mock` here? I think it might be worth asserting in the case that `load_plugins` is modified to eg. returns strings and not exc's for `errors`.

Reply:
Well, it'd be `exc == errors[0]` essentially, but it doesn't make sense to assert that because it's literally what we're telling mock to return. Unless I'm misunderstanding the point?

Reply:
Ah, I misread what's getting monkeypatched, makes sense then :+1: